File: javascript/347-top-K-frequent-elements.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number[]}
 */
var topKFrequent = function (nums, k) {
    const frequency = {};
    for (let i = 0; i < nums.length; i++) {
        if (frequency.hasOwnProperty(nums[i])) frequency[nums[i]] += 1;
        else frequency[nums[i]] = 1;
    }

    const sortedFrequency = Object.keys(frequency).sort((a, b) => frequency[b] - frequency[a]);

    const res = [];
    for (let i = 0; i < k; i++) {
        res.push(sortedFrequency[i]);
    }

    return res;
};

File: python/347-top-K-frequent-elements.py
from collections import defaultdict

class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        res = defaultdict(int)

        for num in nums:
            res[num] += 1

        sorted_res = sorted(res.items(), key=lambda x: x[1], reverse=True)
        return [x[0] for x in sorted_res[:k]]